### ISSUE TITLE
RoI pooling backward assertion failed

### ISSUE THREAD

**Issue by JohanCao** [dmlc/mxnet#2619]
Hi Guys,

I was running the rcnn code in example and I keep get the errors: 
**Assertion `roi_batch_ind >= 0' failed** or **Assertion `roi_batch_ind < batch_size_' failed** during the RoI pooling backward calculation. Has anyone get these errors before? Many thanks!

p.s. the code works fine with config.TRAIN.BATCH_IMAGES = 2 in config.py, I get the errors after change config.TRAIN.BATCH_IMAGES to **1**.

Regards
Johan

**Comment by sxjscience** [comment]
@precedenceguo

**Comment by precedenceguo** [comment]
I cannot produce this error. Would you please elaborate your situation, specifically code modifications if any?

**Comment by JohanCao** [comment]
@precedenceguo  thank you for your reply. I had that error when I debug on CPU local machine. When I run the code on GPU cluster, no errors. Maybe it just not working on CPUs.

However I got another problem. The code works fine for the first several hundreds of iterations. Then it stops due to GPU out of memory (~10GB). I minimized the input image size but not helping. Why the GPU memory usage is unstable? The VOC07 dataset has no considerably large images.

Regards
